docs/src/snap/howto/two-node-ha.md

```diff
@@ -0,0 +1,426 @@
+[Corosync]. After a node failure, the DRBD volume will be mounted on the
+standby node, allowing access to the latest Dqlite database version.
+
+Additional recovery steps are automated and invoked through Pacemaker.
+
+### Prerequisites:
+
+* Please ensure that both nodes are part of the Kubernetes cluster.
+  See the [getting started] and [add/remove nodes] guides.
+* The user associated with the HA service has SSH access to the peer node and
+  passwordless sudo configured. For simplicity, the default "ubuntu" user can
+  be used.
+* We recommend using static IP configuration.
+
+The [two-node-ha.sh script] automates most operations related to the two-node
+HA scenario and is included in the snap.
+
+The first step is to install the required packages:
+
+```
+/snap/k8s/current/k8s/hack/two-node-ha.sh install_packages
+```
+
+### Distributed Replicated Block Device (DRBD)
+
+This example uses a loopback device as DRBD backing storage:
+
+```
+sudo dd if=/dev/zero of=/opt/drbd0-backstore bs=1M count=2000
+```
+
+Ensure that the loopback device is attached at boot time, before Pacemaker
+starts.
+
+```
+cat <<EOF | sudo tee /etc/rc.local
+#!/bin/sh
+mknod /dev/lodrbd b 7 200
+losetup /dev/lodrbd /opt/drbd0-backstore
+EOF
+
+sudo chmod +x /etc/rc.local
+```
+
+Add a service to automatically execute the ``/etc/rc.local`` script.
+
+```
+
+
+cat <<EOF | sudo tee /etc/systemd/system/rc-local.service
+# This unit gets pulled automatically into multi-user.target by
+# systemd-rc-local-generator if /etc/rc.local is executable.
+[Unit]
+Description=/etc/rc.local Compatibility
+Documentation=man:systemd-rc-local-generator(8)
+ConditionFileIsExecutable=/etc/rc.local
+After=network.target
+
+[Service]
+Type=forking
+ExecStart=/etc/rc.local start
+TimeoutSec=0
+RemainAfterExit=yes
+GuessMainPID=no
+
+[Install]
+WantedBy=multi-user.target
+EOF
+
+sudo systemctl enable rc-local.service
+sudo systemctl start rc-local.service
+```
+
+Configure the DRBD block device that will hold the Dqlite data.
+Please ensure that the correct node addresses are used.
+
+```
+# Disable the DRBD service, it will be managed through Pacemaker.
+sudo systemctl disable drbd
+
+HAONE_ADDR=<firstNodeAddress>
+HATWO_ADDR=<secondNodeAddress>
+
+cat <<EOF | sudo tee /etc/drbd.d/r0.res
+resource r0 {
+  on haone {
+    device /dev/drbd0;
+    disk /dev/lodrbd;
+    address ${HAONE_ADDR}:7788;
+    meta-disk internal;
+  }
+  on hatwo {
+    device /dev/drbd0;
+    disk /dev/lodrbd;
+    address ${HATWO_ADDR}:7788;
+    meta-disk internal;
+  }
+}
+EOF
+
+sudo drbdadm create-md r0
+sudo drbdadm status
+```
+
+Create a mount point for the DRBD block device. Non-default mount points
+need to be passed to the ``two-node-ha.sh`` script mentioned above, see the
+script for the full list of configurable parameters.
+
+```
+DRBD_MOUNT_DIR=/mnt/drbd0
+sudo mkdir -p $DRBD_MOUNT_DIR
+```
+
+Run the following *once* to initialize the filesystem.
```

Comment:
Does something bad happen if we run it twice that users should know about?

Reply:
Not really. It's just that unlike the other commands,  this doesn't have to be called twice.